src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can use destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, .......... } = action;
  
    switch (type) {
```